ipfs-6-to-7/migration/migration.go

```diff
@@ -0,0 +1,237 @@
+package mg6
+
+import (
+	"fmt"
+
+	migrate "github.com/ipfs/fs-repo-migrations/go-migrate"
+	log "github.com/ipfs/fs-repo-migrations/stump"
+
+	dshelp "gx/ipfs/QmTmqJGRQfuH8eKWD1FjThwPRipt1QhqJQNZ8MpzmfAAxo/go-ipfs-ds-help"
+	record "gx/ipfs/QmUpttFinNDmNPgFwKN8sZK6BUtBmA68Y4KdSBDXa8t9sJ/go-libp2p-record"
+	dhtpb "gx/ipfs/QmUpttFinNDmNPgFwKN8sZK6BUtBmA68Y4KdSBDXa8t9sJ/go-libp2p-record/pb"
+	ds "gx/ipfs/QmXRKBQA4wXP7xWbFiZsR1GP4HV6wMDQ1aWFxZZ4uBcPX9/go-datastore"
+	proto "gx/ipfs/QmZ4Qi3GaRbjcx28Sme5eMH7RQjGkt8wHxt2a65oLaeFEV/gogo-protobuf/proto"
+	peer "gx/ipfs/QmZoWKhxUmZ2seW4BzX6fJkNR8hh9PsGModr7q171yq2SS/go-libp2p-peer"
+	ci "gx/ipfs/QmaPbCnUMBohSGo3KnxEa2bHqyJVVeEEcwtqJAYxerieBo/go-libp2p-crypto"
+	namesys "gx/ipfs/QmcKwjeebv5SX3VFUGDFa4BNMYhy14RRaCzQP7JN3UQDpB/go-ipfs/namesys"
+	repo "gx/ipfs/QmcKwjeebv5SX3VFUGDFa4BNMYhy14RRaCzQP7JN3UQDpB/go-ipfs/repo"
+	fsrepo "gx/ipfs/QmcKwjeebv5SX3VFUGDFa4BNMYhy14RRaCzQP7JN3UQDpB/go-ipfs/repo/fsrepo"
+	mfsr "gx/ipfs/QmcKwjeebv5SX3VFUGDFa4BNMYhy14RRaCzQP7JN3UQDpB/go-ipfs/repo/fsrepo/migrations"
+	base32 "gx/ipfs/QmfVj3x4D6Jkq9SEoi5n2NmoUomLwoeiwnYz2KQa15wRw6/base32"
+)
+
+type Migration struct{}
+
+func (m Migration) Versions() string {
+	return "6-to-7"
+}
+
+func (m Migration) Reversible() bool {
+	return true
+}
+
+func myKey(r repo.Repo) (ci.PrivKey, error) {
+	cfg, err := r.Config()
+	if err != nil {
+		return nil, err
+	}
+
+	sk, err := cfg.Identity.DecodePrivateKey("passphrase todo!")
+	if err != nil {
+		return nil, err
+	}
+
+	pid, err := peer.IDFromPrivateKey(sk)
+	if err != nil {
+		return nil, err
+	}
+	idCfg, err := peer.IDB58Decode(cfg.Identity.PeerID)
+	if err != nil {
+		return nil, err
+	}
+
+	if pid != idCfg {
+		return nil, fmt.Errorf(
+			"private key in config does not match id: %s != %s",
+			pid,
+			idCfg,
+		)
+	}
+	return sk, nil
+}
+
+func applyForKey(dstore ds.Datastore, k ci.PrivKey) error {
```

Comment:
it does feel a bit odd to be using private keys here instead of just public keys, but its not a huge deal i guess

Reply:
Yeah... but that's what the Keystore/Identity gives me.

Given that I've already built the dist (and linked to it from go-ipfs), I'll leave this as-is unless I have to make other changes.